include/core/tag.h

```diff
@@ -110,6 +110,15 @@ int lf_tag_compare(tag_t tag1, tag_t tag2);
  */
 tag_t lf_delay_tag(tag_t tag, interval_t interval);
 
+/**
+ * Return the latest tag that cannot have an event provided that
+ * - `antitag` does not have an event, and
+ * - if an event occurs at time `x`, then there exists a tag `antitag` such that
+ *   `x = antitag + interval`, where `+` is the usual `+` that we use for tags
+ *   which is neither commutative nor a group operation.
+ */
+tag_t lf_delay_antitag(tag_t antitag, interval_t interval);
```

Comment:
The description makes sense, but I find the terminology confusing. The fact that the given tag doesn't have an event doesn't change anything about the tag, so I don't see why we should be calling it an "antitag." Maybe it would be better to name the method more descriptively? Also, are you use that you meant "latest", not "earliest"?

Reply:
> I don't see why we should be calling it an "antitag."

I can write `absent_tag` to indicate that it is the tag of something that does not happen.

> Also, are you use that you meant "latest", not "earliest"?

Yes, I meant "latest." To find the earliest tag at which nothing can happen is trivial -- that is the tag that is as far as possible into the past (nothing is going to happen in the past). But actually what I wrote was not technically right. I meant "The maximum of the set of all tags at which nothing can happen," which is similar to "latest"

Reply:
Now I'm even more confused :laughing: 

What do you mean with "tags at which nothing can happen"? Something can happen at any tag in the future. The absence of an event in the queue at a particular tag does not rule out that an event with that tag could later be added.

Reply:
> Something can happen at any tag in the future.

The whole design of federated execution is based on the fact that is not true. We have after delays. If nothing happens on one side of an after delay at time $g$, and the delay is $h$ then nothing will happen on the other side of the after delay before time $g + h$.

Reply:
That is true, but but timers and actions can cause events to be inserted in the queue. By "nothing happens" you mean with respect to a particular upstream federate? I'm not trying to be dense, but the more you explain about the use of this function, the easier it will be to come up with more fitting terminology...

Reply:
Nothing will happen **at the port** on the other side of the after delay.

I have no strong opinions about the terminology and would be OK with whatever terminology we choose. It is good to strike a reasonable balance between choosing sensible terminology and spending too much time debating over terminology.

Reply:
@petervdonovan identified and fixed a real bug in the RTI. This should also resolve an issue I had been seeing that the RTI was sending way too many PTAG messages when these were not needed.  I have adjusted the terminology and implementation and believe this is ready to merge (modulo tests passing).